AvalonStudio/AvalonStudio.Extensibility/Utils/FuzzyMatch.cs

```diff
@@ -0,0 +1,448 @@
+                    {
+                        score += lastSegmentScore * LAST_SEGMENT_BOOST;
+                    }
+
+                    if(currentRange.matched && !currentRangeStartedOnSpecial)
+                    {
+                        score -= NOT_STARTING_ON_SPECIAL_PENALTY;
+                    }
+
+                    ranges.Add(currentRange);
+                }
+
+                if(lastMatchIndex + 1 < c)
+                {
+                    ranges.Add(new Range(str.Substring(lastMatchIndex + 1, c-(lastMatchIndex + 1)), false, c > lastSegmentStart));
+                }
+
+                currentRange = null;
+                lastSegmentScore = 0;
+            }
+
+            int numConsecutive = 0;
+
+            void addMatch(Match match)
+            {
+                int c = match.index;
+                int newPoints = 0;
+
+                newPoints += MATCH_POINTS;
+
+                if(c == lastSegmentStart)
+                {
+                    newPoints += BEGINNING_OF_NAME_POINTS;
+                }
+
+                if(score > 0 && lastMatchIndex + 1 == c)
+                {
+                    if(currentRangeStartedOnSpecial)
+                    {
+                        numConsecutive++;
+                    }
+
+                    newPoints += CONSECUTIVE_MATCHES_POINTS * numConsecutive;
+                } else
+                {
+                    numConsecutive = 1;
+                }
+
+                if(match.isSpecial)
+                {
+                    newPoints += SPECIAL_POINTS;
+                }
+
+                score += newPoints;
+
+                if(c >= lastSegmentStart)
+                {
+                    lastSegmentScore += newPoints;
+                }
+
+                if((currentRange != null && !currentRange.matched) || c > lastMatchIndex + 1)
+                {
+                    closeRangeGap(c);
+                }
+
+                lastMatchIndex = c;
+
+                if(currentRange == null)
+                {
+                    currentRange = new Range(str[c].ToString(), true, false);
+                    currentRangeStartedOnSpecial = (match.isSpecial);
+                } else
+                {
+                    currentRange.text += str[c];
+                }
+            }
+
+            for(matchCounter = 0; matchCounter < matchList.Count; matchCounter++)
+            {
+                Match match = matchList[matchCounter];
+                addMatch(match);
+            }
+
+            closeRangeGap(str.Length);
+
+            int lengthPenalty = (int) (-1 * Math.Round(str.Length * DEDUCTION_FOR_LENGTH));
+
+            score = score + lengthPenalty;
+
+            return (ranges, score);
+        }
+
+        public class SearchResult
+        {
+            public SearchResult(string label)
+            {
+                this.label = label;
+            }
+
+            public string label;
```

Comment:
basically you only have to capitalize and add {get; set;} it costs nothing performance wise, and we maintain accepted .net practices. Performance argument might only be valid for C code.